sos/plugins/samba.py

```diff
@@ -33,6 +33,10 @@ def setup(self):
             "/var/log/samba/log.winbindd-dc-connet",
             "/var/log/samba/log.wb-*"
         ])
+
+        if self.get_option("all_logs"):
+            self.add_copy_spec("/var/log/samba/")
+
```

Comment:
What I'd suggest is something along the lines of:

```python
    self.add_copy_spec([
        "/etc/samba/smb.conf",
        "/etc/samba/lmhosts"
    ])
    log_size = self.get_option("log_size")
    if self.get_option("all_logs"):
        self.add_copy_spec_limit("/var/log/samba/*", sizelimit=log_size)
    else:
        self.add_copy_spec_limit([
            "/var/log/samba/log.smbd",
            "/var/log/samba/log.nmbd",
            "/var/log/samba/log.winbindd",
            "/var/log/samba/log.winbindd-idmap",
            "/var/log/samba/log.winbindd-dc-connet",
            "/var/log/samba/log.wb-*"
        ], sizelimit=log_size)
```

That way the `--log-size` and `--all-logs` options can be freely mixed and matched.